### PR TITLE
Include .mjs files in JavaScript MIME

### PR DESCRIPTION
**Pull request by bmeck**
JavaScript modules will be using a new file extension of .mjs as
specified in the [Node EP](https://github.com/nodejs/node-eps/blob/master/002-es-modules.md).
To allow such files to be use both by the web and by Node, this
PR serves the file extension with the JavaScript MIME, as recommended
by the upcoming [Internet Draft](https://datatracker.ietf.org/doc/draft-bfarias-javascript-mjs/).

### CODE REVIEW

**Comment by mathiasbynens** [comment]
👍  We should definitely add this, to make it easy to deploy the same code in Node.js and on the web.

**Comment by mathiasbynens** [comment]
@bmeck Are you up for making those additional changes? If not, other contributors can step in.

**Comment by bmeck** [comment]
@mathiasbynens missed the comment, ran build script and pushed

**Comment by mathiasbynens** [comment]
@bmeck Thanks, but still more changes are needed. Please take a look at https://github.com/h5bp/server-configs-apache/blob/master/dist/.htaccess and search for `js` (or `js|`) — we need to add `mjs` for pretty much every instance you find (by indeed editing the source files, and then running the build script).

**Comment by bmeck** [comment]
@mathiasbynens wait, wouldn't `dist` be destroyed when using a build script?

**Comment by mathiasbynens** [comment]
Yeah, you’re not supposed to edit `dist/` directly — I meant searching `dist/.htaccess` is just an easy way to find all lines that need changing.

File: src/internet_explorer/x-ua-compatible.conf [hunk 1]
```diff
@@ -23,7 +23,7 @@
     # the `X-UA-Compatible` response header should be send only for
     # HTML documents and not for the other resources.
 
-    <FilesMatch "\.(appcache|atom|bbaw|bmp|crx|css|cur|eot|f4[abpv]|flv|geojson|gif|htc|ico|jpe?g|js|json(ld)?|m4[av]|manifest|map|mp4|oex|og[agv]|opus|otf|pdf|png|rdf|rss|safariextz|svgz?|swf|topojson|tt[cf]|txt|vcard|vcf|vtt|webapp|web[mp]|webmanifest|woff2?|xloc|xml|xpi)$">
+    <FilesMatch "\.(appcache|atom|bbaw|bmp|crx|css|cur|eot|f4[abpv]|flv|geojson|gif|htc|ico|jpe?g|js|json(ld)?|m4[av]|manifest|map|mjs|mp4|oex|og[agv]|opus|otf|pdf|png|rdf|rss|safariextz|svgz?|swf|topojson|tt[cf]|txt|vcard|vcf|vtt|webapp|web[mp]|webmanifest|woff2?|xloc|xml|xpi)$">
```
Comment (mathiasbynens) [inline src/internet_explorer/x-ua-compatible.conf]: This is super minor, but let’s go with `m?js` (similar to `jpe?g`). More obvious that way IMHO

**Comment by mathiasbynens** [comment]
Amended your commits and resolved the merge conflicts. Will merge once CI build passes.

**Pull request merged**